src/diepvries/template_sql/satellite_dml.sql

```diff
@@ -1,64 +1,65 @@
 MERGE INTO {target_schema}.{target_table} AS satellite
   USING (
-    WITH
-      filtered_staging AS (
-        SELECT DISTINCT
-          staging.{hashkey_field},
-          staging.{staging_hashdiff_field},
-          staging.{record_start_timestamp},
-          staging.{record_source}
-          {staging_descriptive_fields}
-        FROM {staging_schema}.{staging_table} AS staging
-          CROSS JOIN (
-                       SELECT
-                         MAX({record_start_timestamp}) AS max_r_timestamp
-                       FROM {target_schema}.{target_table}
-                     ) AS max_satellite_timestamp
-        WHERE staging.{record_start_timestamp} >= COALESCE(max_satellite_timestamp.max_r_timestamp, '1970-01-01 00:00:00')
-      ),
-      staging_satellite_affected_records AS (
-        /* Records that will be inserted (don't exist in target table or exist
-          in the target table but the hashdiff changed). As the r_timestamp is fetched
-          from the staging table, these records will always be included in the
-          WHEN NOT MATCHED condition of the MERGE command. */
-        SELECT
-          staging.{hashkey_field},
-          staging.{staging_hashdiff_field},
-          staging.{record_start_timestamp},
-          staging.{record_source}
-          {staging_descriptive_fields}
-        FROM filtered_staging AS staging
-          LEFT OUTER JOIN {target_schema}.{target_table} AS satellite
-                          ON (staging.{hashkey_field} = satellite.{hashkey_field}
+        WITH
+          filtered_staging AS (
+          SELECT DISTINCT
+            staging.{hashkey_field},
+            staging.{staging_hashdiff_field},
+            staging.{record_start_timestamp},
+            staging.{record_source}
+            {staging_descriptive_fields}
+          FROM {staging_schema}.{staging_table} AS staging
+          WHERE NOT EXISTS (
+                           SELECT
+                             1
+                           FROM {target_schema}.{target_table} AS satellite
+                           WHERE staging.{hashkey_field} = satellite.{hashkey_field}
```

Comment:
```suggestion
                           WHERE {satellite_driving_key_condition}
```
Can we also use this expression like in the other file, or not?

Reply:
Nope. The conditions are different as this is a regular satellite hence the condition is always the same, consistently a join by hashkey. IN the effectivity satellite we have an expression that is dynamic with as many fields in the join as fields in the driving key.

We could define a `{satellite_hashkey_condition}` but would make the code even more difficult to "visually parse" IMO

Reply:
Oh I understand 👍 All good!